data/helpers.d/systemd

```diff
@@ -133,7 +133,7 @@ ynh_systemd_action() {
         for i in $(seq 1 $timeout)
         do
             # Read the log until the sentence is found, that means the app finished to start. Or run until the timeout
-            if grep --quiet "$line_match" "$templog"
+            if grep --extended-regexp --quiet "$line_match" "$templog"
```

Comment:
I looked to apps that use ynh_systemd_action and didn't see anything which could break.
I you have some apps in mind, we can test.

Reply:
By example in [pgadmin](https://github.com/YunoHost-Apps/pgadmin_ynh/blob/testing/scripts/install#L120):
```bash
➜  /tmp echo "WSGI app 0 (mountpoint='/asdf/asdf')" > test
➜  /tmp grep "WSGI app 0 (mountpoint='/asdf/asdf')" test   
WSGI app 0 (mountpoint='/asdf/asdf')
➜  /tmp grep -E "WSGI app 0 (mountpoint='/asdf/asdf')" test
➜  /tmp 
```

Reply:
Indeed, the parenthesis don't like -E, you can replace them by a .
It works in both situation.

Reply:
Well, the question is not to fix the app just adding a `\` before fix the issue. The problem is that it might break many apps when we will merge this.

Reply:
Many I don't think so...
I can have another look to it, but to fix fail2ban sounds like a pretty good idea...
Also, to have proper regex in that helper may help a lot when trying to find the correct line.

Reply:
Oh shit you were right !!!
I just had a look to all apps using this helper with that option, and try all strings with more than just standard characters with `grep` and `grep -E`.
It appears it break many apps !!!

Here the huge list of apps concerned by the bug I was about to merge :flushed: !
```
pgadmin_ynh
```
... :unamused:

Won't say I'm surprised....
So what should we do ? Should we replace parenthesis by dots and check the [[:digit:]]* into **your** app ?
Or should we close that PR, delete the branch, and forget forever that madness I was about to do ?

Reply:
> Should we replace parenthesis by dots and check the [[:digit:]]* into your app ?

Maybe we can just add an argument for this helper to specify which pattern syntax we use. It could be `extended`, `fixed`, `perl`, `basic` (by default).

> Or should we close that PR, delete the branch, and forget forever that madness I was about to do ?

No because I also think that fail2ban make some issue that should be fixed.